cmd/SplitFFISSpreadsheet/sheet.go

```diff
@@ -0,0 +1,217 @@
+				// assume it's a bill and continue
+				if !cfdaRegex.MatchString(cell) {
+					bill = cell
+					continue rowLoop
+				}
+			}
+
+			// Populate opportunity based on an assumed format
+			// where colIndex is a column (zero is A, 1 is B, etc.)
+			switch colIndex {
+			case 0:
+				opportunity.CFDA = cell
+			case 1:
+				opportunity.OppTitle = cell
+			case 2:
+				opportunity.Agency = cell
+			case 3:
+				// If estimated funding is N/A, assume 0
+				if cell == "N/A" {
+					continue
+				}
+
+				num, err := strconv.ParseInt(cell, 10, 64)
+				// If we can't parse the funding amount, just skip the column
+				if err != nil {
+					log.Warn(logger, "Error parsing estimated funding", err)
+					sendMetric("spreadsheet.cell_parsing_errors", 1)
+					continue
+				}
+				opportunity.EstimatedFunding = num
+			case 4:
+				opportunity.ExpectedAwards = cell
+			case 5:
+				opportunity.OppNumber = cell
+
+				// cellAxis (eg. A4) is used to get a hyperlink for a cell. We
+				// need to increment the index because Excel is not zero-indexed
+				cellAxis, err := excelize.CoordinatesToCellName(colIndex+1, rowIndex+1)
+				if err != nil {
+					log.Warn(logger, "Error parsing cell axis for grant ID", err)
+					sendMetric("spreadsheet.cell_parsing_errors", 1)
+					continue
+				}
+
+				hasLink, target, err := xlFile.GetCellHyperLink(sheet, cellAxis)
+				if err != nil {
+					// log this, it is not worth aborting the whole extraction for
+					log.Warn(logger, "Error getting cell hyperlink for grant ID", err)
+					sendMetric("spreadsheet.cell_parsing_errors", 1)
+					continue
+				}
+
+				// If we have a link, parse the URL for the opportunity ID which
+				// is the only way to get it from the spreadsheet
+				if hasLink {
+					url, err := url.Parse(target)
+					if err != nil {
+						log.Warn(logger, "Error parsing link for grant ID", err)
+						sendMetric("spreadsheet.cell_parsing_errors", 1)
+						continue
+					}
+
+					// The opportunity ID should be a < 20 digit numeric value
+					oppID, err := strconv.ParseInt(url.Query().Get("oppId"), 10, 64)
+					if err != nil {
+						log.Warn(logger, "Error parsing opportunity ID", err)
+						sendMetric("spreadsheet.cell_parsing_errors", 1)
+						continue
+					}
+
+					opportunity.GrantID = oppID
+				}
+			case 6:
+				opportunity.Eligibility.State = parseEligibility(cell)
+			case 7:
+				opportunity.Eligibility.Local = parseEligibility(cell)
+			case 8:
+				opportunity.Eligibility.Tribal = parseEligibility(cell)
+			case 9:
+				opportunity.Eligibility.HigherEducation = parseEligibility(cell)
+			case 10:
+				opportunity.Eligibility.NonProfits = parseEligibility(cell)
+			case 11:
+				opportunity.Eligibility.Other = parseEligibility(cell)
+			case 12:
+				// If we fail to parse the date, just skip the column
+				// and not the whole row
+				t, err := time.Parse("1-2-06", cell)
+				if err != nil {
+					log.Warn(logger, "Error parsing date", err)
+					continue
+				}
```

Comment:
Worth adding `sendMetric("spreadsheet.cell_parsing_errors", 1)` here as well?

Reply:
ah yes missed that one!